pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/design/AbstractCounterCheckRule.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.apex.rule.design;
+
+import static net.sourceforge.pmd.properties.constraints.NumericConstraints.positive;
+
+import java.lang.reflect.Modifier;
+
+import net.sourceforge.pmd.lang.apex.ast.AbstractApexNodeBase;
+import net.sourceforge.pmd.lang.apex.ast.ApexNode;
+import net.sourceforge.pmd.lang.apex.rule.AbstractApexRule;
+import net.sourceforge.pmd.lang.rule.internal.CommonPropertyDescriptors;
+import net.sourceforge.pmd.properties.PropertyDescriptor;
+
+
+/**
+ * Abstract class for rules counting some integer metric on some node.
+ *
+ * @author Clément Fournier
+ * @since 6.7.0
+ */
+abstract class AbstractCounterCheckRule<T extends ApexNode<?>> extends AbstractApexRule {
```

Comment:
package-private? or shall me move it to an internal package such as `net.sourceforge.pmd.lang.apex.rule.internal`?

Reply:
I think yours is a good idea, there's no reason to confine this to the design category